cypress/e2e/Admin/results/TableByQuestion.cy.js

```diff
@@ -167,23 +158,53 @@ describe('Table by Question', () => {
       });
     });
   });
-});
 
-/**
- * Helper function to convert a color from hex to rgb
- * Needed because the color in css property is in rgb, but the primary color is in hex format
- *
- * @param hexColor the color in hexadecimal format
- * @returns {`rgb(${number}, ${number}, ${number})`} The color as rgb definition
- */
-const hexToRGB = (hexColor) => {
-  const bigint = parseInt(hexColor.slice(1), 16);
-  const r = (bigint >> 16) & 255;
-  const g = (bigint >> 8) & 255;
-  const b = bigint & 255;
+  it('click on user redirect us to corresponding table by user', () => {
+    cy.setupResultTablesByQuestionForCheck(APP_SETTINGS_2, APP_DATA_2);
 
-  return `rgb(${r}, ${g}, ${b})`;
-};
+    const rgbBorderColor = hexToRGB(theme.palette.primary.main);
+
+    const fstQuestionId = getSettingsByName(
+      APP_SETTINGS_2,
+      APP_SETTING_NAMES.QUESTION_LIST
+    )[0].data.list[0];
+    const fstQuestion = getSettingsByName(
+      APP_SETTINGS_2,
+      APP_SETTING_NAMES.QUESTION
+    ).find((setting) => setting.id === fstQuestionId).data.question;
+
+    const users = new Set(APP_DATA_2.map((data) => data.memberId));
+    let i = 0;
+    for (const user of users) {
```

Comment:
You can use this kind of `for` loop to avoid your `i` variable.
```
for (const [index, value] of [1, 2, 3, 4, 5].entries())
```

Reply:
Thanks for this tips, it's way cleaner.